### PR TITLE
ApiTests. refactoring apiSpace suite

### PR DESCRIPTION
**Pull request by ScharfViktor**
- "creating project space" moved to background
- many scenarios are combined into one
- added test re-sharing to the group

### CODE REVIEW

**Comment by ownclouders** [comment]
:boom: Acceptance test [<strong>localApiTests-apiSpaces-ocis</strong>](https://drone.owncloud.com/owncloud/ocis/13136/38/1) failed. Further test are cancelled...

File: tests/acceptance/features/apiSpaces/restoreSpaces.feature [hunk 1]
```diff
@@ -14,58 +14,50 @@ Feature: Restoring space
...
+    And user "Alice" has disabled a space "restore a space"
+    When user "Alice" restores a disabled space "restore a space"
+    Then for user "<user>" the space "restore a space" should contain these entries:
+      | test.txt   |
+      | mainFolder |
+    Examples:
+      | user  |
+      | Alice |
+      | Brian |
+      | Bob   |
```
Comment (individual-it) [inline tests/acceptance/features/apiSpaces/restoreSpaces.feature:41]: I think we can save time by not having examples but multiple Then steps
```suggestion
    Then for user "Alice" the space "restore a space" should contain these entries:
      | test.txt   |
      | mainFolder |
    And for user "Brian" the space "restore a space" should contain these entries:
      | test.txt   |
      | mainFolder |
    And for user "Bob" the space "restore a space" should contain these entries:
      | test.txt   |
      | mainFolder |      
```

File: tests/acceptance/features/apiSpaces/shareSubItemOfSpace.feature [hunk 2]
```diff
@@ -18,64 +13,41 @@ Feature: Share a file or folder that is inside a space
...
-  Scenario Outline: An user-owner can share a folder inside project space to another user with role:
-    When user "Alice" shares the following entity "folder" inside of space "share sub-item" with user "Brian" with role "<role>"
+  Scenario Outline: A manager of the space can share an entity inside project space to another user with role:
+    When user "Alice" shares the following entity "<entity>" inside of space "share sub-item" with user "Brian" with role "<role>"
     Then the HTTP status code should be "200"
     And the OCS status code should be "200"
     And the OCS status message should be "OK"
-    And user "Brian" accepts share "/folder" offered by user "Alice" using the sharing API
-    And as "Brian" folder "Shares/folder" should exist
+    And user "Brian" accepts share "/<entity>" offered by user "Alice" using the sharing API
```
Comment (individual-it) [inline tests/acceptance/features/apiSpaces/shareSubItemOfSpace.feature:27]: this sounds like a When step

File: tests/acceptance/features/apiSpaces/shareSubItemOfSpace.feature [hunk 3]
```diff
@@ -18,64 +13,41 @@ Feature: Share a file or folder that is inside a space
...
+  Scenario Outline: An user participant of the project space with manager role can share an entity to another user
     Given user "Alice" has shared a space "share sub-item" to user "Brian" with role "manager"
-    When user "Brian" shares the following entity "folder/file.txt" inside of space "share sub-item" with user "Bob" with role "<role>"
+    When user "Brian" shares the following entity "<entity>" inside of space "share sub-item" with user "Bob" with role "<role>"
     Then the HTTP status code should be "200"
     And the OCS status code should be "200"
     And the OCS status message should be "OK"
-    And user "Bob" accepts share "/file.txt" offered by user "Brian" using the sharing API
-    Then as "Bob" file "Shares/file.txt" should exist
+    And user "Bob" accepts share "/<entity>" offered by user "Brian" using the sharing API
```
Comment (individual-it) [inline tests/acceptance/features/apiSpaces/shareSubItemOfSpace.feature:43]: When step?

**Review by individual-it — CHANGES_REQUESTED** [review]

File: tests/acceptance/features/bootstrap/SpacesContext.php [hunk 4]
```diff
@@ -1034,23 +1016,60 @@ public function jsonRespondedShouldContain(
 	}
 
 	/**
-	 * @Then /^the json responded should contain a space "([^"]*)" granted to "([^"]*)" with role "([^"]*)"$/
+	 * @Then /^the user "([^"]*)" should have a space called "([^"]*)" with these key and value pairs:$/
+	 * @Then /^the user "([^"]*)" should have a space called "([^"]*)" (?:owned by|granted to) "([^"]*)" with these key and value pairs:$/
+	 * @Then /^the user "([^"]*)" should have a space called "([^"]*)" (?:with description file|with space image) "([^"]*)" with these key and value pairs:$/
+	 * @Then /^the user "([^"]*)" should have a space called "([^"]*)" (?:owned by|granted to) "([^"]*)" (?:with description file|with space image) "([^"]*)" with these key and value pairs:$/
```
Comment (individual-it) [inline tests/acceptance/features/bootstrap/SpacesContext.php:1022]: enjoying regex :-)

**Review by individual-it — APPROVED** [review]

**Pull request merged**